modules/bibauthorid/lib/bibauthorid_dbinterface.py

```diff
@@ -2290,8 +2300,14 @@ def _get_inspire_id_of_author(pid):  # get_inspire_ids_by_pids
     @return: Inspire external identifier
     @rtype: tuple ((str),)
     '''
-    return _select_from_aidpersoniddata_where(select=['data'], pid=pid, tag='extid:INSPIREID')
-
+    result =  _select_from_aidpersoniddata_where(select=['data'], pid=pid, tag='extid:INSPIREID')
+    ##WRONG exception here
+    if result:
+        if len(result) > 1:
+            from invenio.bibauthorid_hoover_exceptions import MultipleIdsOnSingleAuthorException
+            raise MultipleIdsOnSingleAuthorException('Conflict in aidPERSONIDDATA', pid, 'INSPIREID', result)
+        return result[0][0]
```

Comment:
@petros-ioannidis here you are changing the return type of the function! Better keep it like that and raise the exception in hoover instead.